lib/generators/declare_schema/migration/migrator.rb

```diff
@@ -478,17 +478,25 @@ def change_indexes(model, old_table_name, to_rename)
           [Array(change_primary_key) + drop_indexes + add_indexes]
         end
 
-        def index_changes_due_to_column_renames(indexes_to_drop, indexes_to_add, to_rename)
-          indexes_to_drop.each_with_object([[], []]) do |index_to_drop, (renamed_indexes_to_drop, renamed_indexes_to_add)|
+        # @return [Array<Array<IndexDefinition>>] pair of arrays of indexes that changed solely due to column renames...not any settings changes.
```

Comment:
The contract here was the source of the bug. The calling code needs to know index changes that _only_ happened because of column renames, since those are implied by renaming the columns and don't need a separate schema change.